packages/extension/src/background/preAuthorizationMessaging.ts

```diff
@@ -9,7 +9,7 @@ import { HandleMessage } from "./background"
 import { openUi } from "./openUi"
 
 preAuthorizeStore.subscribe(async (_, changeSet) => {
-  const removed = difference(changeSet.oldValue ?? [], changeSet.newValue ?? [])
+  const removed = differenceWith(changeSet.oldValue ?? [], changeSet.newValue ?? [], isEqual)
```

Comment:
The purpose of this code snippet is to notify the dapp to disconnect when preauthorization is removed from storage. But the `difference` function doesn't make a deep equal comparison on objects. This will cause old connections to be disconnected when new ones are established.